src/lib/react-query/projects.ts

```diff
@@ -120,11 +119,16 @@ export function projectByIdQueryOptions({
 }: {
 	clientApi: MapeoClientApi
 	projectId: string
-}) {
+}): UseSuspenseQueryOptions<
```

Comment:
Personally would avoid the explicit type annotation here but think this should use `UseQueryOptions` instead, which will allow it to be used by both `useQuery` and `useSuspenseQuery`

Reply:
ah seeing that causes a type issue. The return type of `queryOptions` is smarter than `UseQueryOptions` and `UseSuspenseQueryOptions`. Think it'd still be preferable to avoid this explicit return type in that case.

Reply:
Any ideas for solving this welcome! I'm not sure how to proceed on this one.

Reply:
> I think this changed because rpc-reflector is now ESM. It might have been an issue before, but TS did not spot it.

I'm wondering if it's something related to tshy - or more generally, the separate creation of esm and cjs modules that tshy does. If it were just a normal typescript issue, we'd see this error when running `npm run types` too, but we don't.

Will require a harder think about how to address. I don't have any better suggestions/ideas at the moment.

Reply:
to clarify, not necessarily blaming tshy. could very well be that it's surfacing a legitimate issue since tshy is mostly wrapping tsc. just noting that the error only shows up when doing the tshy-related work

Reply:
The error is not tshy - it happens with `tsc` too. It is an error related to `emit: true` - that's why we don't see it with a type check. It's an error related to how things are exported when importing from another module. When returns are implicit, the types of any dependencies are not included. Explicit returns ensures that the types of any dependencies are included correctly in the outputted code. At least, that's how I understand the issue.

Reply:
> The error is not tshy - it happens with `tsc` too. It is an error related to `emit: true` - that's why we don't see it with a type check. It's an error related to how things are exported when importing from another module. When returns are implicit, the types of any dependencies are not included. Explicit returns ensures that the types of any dependencies are included correctly in the outputted code. At least, that's how I understand the issue.

ah that makes sense. 

> Maybe don't create cjs? We don't need it, and I see a lot about how it can create issues,

Is it not needed for things to work with Metro? Maybe what you've been doing with the other modules i.e. defining `main` and `types` fields is all that's needed in that case?

Reply:
no, issues remain with Jest, it was a pain getting it to work with the new rpc-reflector. They still haven't fully added ESM support...